### PR TITLE
feat: always update development images

### PR DESCRIPTION
**Pull request by theseion**
- new option to force download of an image and override the cache
- never use the cached image for in-development versions (*-trunk | *-alpha)

### CODE REVIEW

**Comment by theseion** [comment]
Note: this change is necessary due to a fix a couple of months back that fixed the caching behaviour (#604).

File: run.sh [hunk 1]
```diff
@@ -344,6 +360,10 @@ parse_options() {
       fi
       shift 2
       ;;
+    -u | --update-image)
```
Comment (fniephaus) [inline run.sh:363]: `--update-image` sounds like it triggers an in-image update. Maybe something like `--refresh-cache` or `--overwrite-cache`?

Now that I think about this, maybe trunk/alpha images should never be cached, or always overwritten. For dev purposes, we could also provide the inverse option, something like `--force-cache`.

BTW: we also already have `--clean` for cleaning the cache.

Comment (theseion) [reply run.sh:363]: Yes, those sound better.

> Now that I think about this, maybe trunk/alpha images should never be cached, or always overwritten. For dev purposes, we could also provide the inverse option, something like `--force-cache`.

That was the main intent of this option. `--clean` clears the entire cache, I simply wanted to use the latest image for trunk / alpha.

File: pharo/run.sh [hunk 2]
```diff
@@ -278,14 +278,20 @@ pharo::prepare_image() {
...
-    pushd "${target}" > /dev/null
+  fi
+  if ! is_file "${target}"/*.image; then
+    pushd "${target}" >/dev/null
     fold_start download_image "Downloading ${smalltalk_name} image..."
       download_file "${pharo_image_url}" "${pharo_zeroconf}"
       retry 3 "bash ${pharo_zeroconf}"
     fold_end download_image
-    popd > /dev/null
+    popd >/dev/null
```
Comment (fniephaus) [inline pharo/run.sh:294]:
```suggestion
    popd > /dev/null
```

File: run.sh [hunk 3]
```diff
@@ -280,12 +280,34 @@ validate_configuration() {
...
 
+################################################################################
+# Set options that depend on the contex, i.e., the input values and selections
+# performed so far.
+################################################################################
+set_context_options() {
+    if [ "${config_force_cache}" = "true" ]; then
+      print_info "Forcing cache use"
+      config_overwrite_cache="false"
+      return
```
[the remark this answers is not in the document]
Comment (theseion) [reply run.sh:291]: Like this?

[the remark this answers is not in the document]
Comment (fniephaus) [reply run.sh]: Yes, thanks!

**Review by fniephaus — APPROVED** [review]
LGTM

**Pull request merged**